packages/tpetra/core/src/Tpetra_MultiVector_def.hpp

```diff
@@ -3921,6 +3921,7 @@ namespace Tpetra {
       ProfilingRegion regionGemm ("Tpetra::MV::multiply-call-gemm");
       KokkosBlas::gemm (&ctransA, &ctransB, alpha_IST, A_sub, B_sub,
                         beta_local, C_sub);
+      Kokkos::fence();
```

Comment:
What's actually broken after this?  Is it the code immediately below, like the `::Tpetra::deep_copy` or `reduce`, or is it code not in this function?  My guess is that it's the `reduce`, in which case the `fence` should go right above the `reduce` call (that would be nicer, since it would skip the fence in some cases).  If it's Belos itself, then we should fence in the `Belos::MultiVecTraits` specialization.